Dima/netflix-roulette/src/Components/Sort/Sort.tsx

```diff
@@ -6,25 +6,50 @@ import * as H from 'history'
 interface ISortProps {
   handleSearchChange( { sortBy}: { sortBy: string } ): void,
   numberOfFilms: number,
-  location: H.Location
+  location: H.Location,
+  history: H.History
 };
 
 interface ISortState {
   sortBy: string
 };
 
+enum ISortByParams {
+  ReleaseDate = 'release_date',
+  VoteAverage = 'vote_average'
+}
+
 export default class Sort extends React.Component<ISortProps, ISortState>{
   state: ISortState = {
-    sortBy: 'release_date'
+    sortBy: ISortByParams.ReleaseDate
   };
 
-  componentDidMount() {
-    const URLData = parse(this.props.location.search) as {
-      sortBy: string;
-    };
-    let { sortBy } = URLData;
-    sortBy && this.setState({ sortBy });
-  };
+  // componentDidMount() {
+  //   console.log(this.state.sortBy)
+  //   const URLData = parse(this.props.location.search) as {
+  //     sortBy: string;
+  //   };
+  //   let { sortBy } = URLData;
+  //   if(!sortBy) {
+  //     this.setState({ sortBy: ISortByParams.ReleaseDate });
+  //   }
+  // };
+
+  componentDidUpdate(prevProps: ISortProps ) {
+    if(this.props.location !== prevProps.location && this.props.history.action !== 'PUSH') {
+      const URLData = parse(this.props.location.search) as {
+        sortBy: string;
+      };
+      let { sortBy } = URLData;
+      if(sortBy === ISortByParams.ReleaseDate) {
+        this.setState({ sortBy });
+      } else if(sortBy === ISortByParams.VoteAverage) {
+        this.setState({ sortBy });
+      } else {
+        this.setState({ sortBy: ISortByParams.ReleaseDate });
+      }
+    }
```

Comment:
Logic looks unnecessarily complicated. Make it easier, cleaner, remove redundant IFs